aspnetcore/fundamentals/configuration.md

```diff
@@ -5,162 +5,981 @@ ms.author: riande
+           return Content($"subOption1 = {subOption1}, subOption2 = {subOption2}");
+       }
+   }
+
+   ````
+
+`subOption1 = subvalue1_from_json, subOption2 = 200` is returned.
+
+<a name=in-memory-provider></a>
+
+## In-memory provider and binding to a POCO class
+
+The following sample shows how to use the in-memory provider and bind to a class:
+
+[!code-csharp[Main](configuration/sample/src/InMemory/Program.cs)]
+
+<!-- literal_block {"xml:space": "preserve", "language": "none", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/InMemory/Program.cs", "highlight_args": {"linenostart": 1}, "names": []} -->
+
+````none
+
+   using Microsoft.Extensions.Configuration;
+   using System;
+   using System.Collections.Generic;
+
+   // Add NuGet  <package id="Microsoft.Extensions.Configuration.Binder"
+// rm
+   public class Program
+   {   
+       static public IConfigurationRoot Configuration { get; set; }
+       public static void Main(string[] args = null)
+       {
+           var dict = new Dictionary<string, string>
+               {
+                   {"Profile:MachineName", "Rick"},
+                   {"App:MainWindow:Height", "11"},
+                   {"App:MainWindow:Width", "11"},
+                   {"App:MainWindow:Top", "11"},
+                   {"App:MainWindow:Left", "11"}
+               };
+
+           var builder = new ConfigurationBuilder();
+           builder.AddInMemoryCollection(dict);
+           Configuration = builder.Build();
+           Console.WriteLine($"Hello {Configuration["Profile:MachineName"]}");
+
+           var window = new MyWindow();
+           Configuration.GetSection("App:MainWindow").Bind(window);
+           Console.WriteLine($"Left {window.Left}");
+       }
+   }
+   ````
+
+[!code-csharp[Main](configuration/sample/src/InMemory/MyWindow.cs)]
+
+<!-- literal_block {"xml:space": "preserve", "language": "none", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/InMemory/MyWindow.cs", "highlight_args": {"linenostart": 1}, "names": []} -->
 
-The `Index` view displays the configured options:
+````none
 
-![image](configuration/_static/index-view.png)
+   public class MyWindow
+   { // rm
+       public int Height { get; set; }
+       public int Width { get; set; }
+       public int Top { get; set; }
+       public int Left { get; set; }
+   }
+   ````
 
-You configure options using the `Configure<TOptions>` extension method. You can configure options using a delegate or by binding your options to configuration:
+Configuration values are not limited to scalars. You can retrieve POCO objects or even entire object graphs. The following sample shows how to bind to the `MyWindow` class and use the options pattern with a ASP.NET Core MVC app:
```

Comment:
> Configuration values are not limited to scalars.

But at the Configuration level, they are limited to scalars: they are actually just strings. I am not sure saying otherwise helps. It is only binding that builds on top of that to enable constructing objects.